### PR TITLE
Wrap wad and ray in structs

### PR DESCRIPTION
**Pull request by swapnilraj**

### CODE REVIEW

File: contracts/src/l2/rewaave/math/wad_ray_math.cairo [hunk 1]
```diff
@@ -49,176 +57,196 @@ func uint256_max() -> (max : Uint256):
...
-        if b.high + b.low == 0:
+        if b.ray.high + b.ray.low == 0:
             assert 1 = 0
         end
     end
 
     let (RAY_UINT) = ray()
 
-    let (aRAY, overflow) = uint256_mul(a, RAY_UINT)
+    let (aRAY, overflow) = uint256_mul(a.ray, RAY_UINT.ray)
```
Comment (JorikSchellekens) [inline contracts/src/l2/rewaave/math/wad_ray_math.cairo:229]: Do these not add a bunch of `RAY + 0`s to the assembled version?

Comment (JorikSchellekens) [reply contracts/src/l2/rewaave/math/wad_ray_math.cairo:229]: Or is the compiler smart enough to figure that out?

**Comment by JorikSchellekens** [comment]
Unfortunately this change doesn't really make it any cleaner or more type safe. Although it does make the expectations a little more explicit

**Comment by swapnilraj** [comment]
> Unfortunately this change doesn't really make it any cleaner or more type safe. Although it does make the expectations a little more explicit

Yea I agree, I think it makes the claimable library a bit safer but since there needs to be an explicit conversion from uint256 to ray/wad; there is a still a chance that the user writes it wrong

**Review by JorikSchellekens — APPROVED** [review]

**Pull request merged**